consumer/session/session_storage_test.go

```diff
@@ -68,49 +70,131 @@ func TestSessionStorageGetAllReturnsError(t *testing.T) {
+			ProviderCountry:   "MU",
+			Started:           time.Date(2020, 4, 1, 10, 11, 12, 0, time.UTC),
+			Status:            "Completed",
+			Updated:           time.Date(2020, 4, 1, 12, 0, 0, 0, time.UTC),
+			DataStats:         mockStats,
+			Invoice:           crypto.Invoice{},
+		},
+		storer.UpdateCalled,
+	)
 }
 
-func TestSessionStorageConsumeEventConnectedOK(t *testing.T) {
+func TestSessionStorage_consumeEventConnectedOK(t *testing.T) {
 	storer := &StubSessionStorer{}
 
 	storage := NewSessionStorage(storer)
 	storage.consumeSessionEvent(connection.AppEventConnectionSession{
 		Status:      connection.SessionCreatedStatus,
 		SessionInfo: mockSession,
 	})
-	assert.True(t, storer.SaveCalled)
+	assert.Equal(
+		t,
+		&History{
+			SessionID:         node_session.ID("sessionID"),
+			ConsumerID:        identity.FromAddress("consumerID"),
+			AccountantAddress: "0x00000000000000000000000000000000000000AC",
+			ProviderID:        identity.FromAddress("providerID"),
+			ServiceType:       "serviceType",
+			ProviderCountry:   "MU",
+			Started:           time.Date(2020, 4, 1, 10, 11, 12, 0, time.UTC),
+			Status:            "New",
+			Updated:           time.Time{},
+			DataStats:         connection.Statistics{},
+			Invoice:           crypto.Invoice{},
+		},
+		storer.SaveCalled,
+	)
+}
+
+func TestSessionStorage_consumeSessionSpendingEvent(t *testing.T) {
+	storer := &StubSessionStorer{}
+
+	storage := NewSessionStorage(storer)
+	storage.timeGetter = func() time.Time {
+		return time.Date(2020, 4, 1, 12, 0, 0, 0, time.UTC)
+	}
+	storage.consumeSessionEvent(connection.AppEventConnectionSession{
+		Status:      connection.SessionCreatedStatus,
+		SessionInfo: mockSession,
+	})
+
+	storage.consumeSessionSpendingEvent(event.AppEventInvoicePaid{
+		ConsumerID: identity.FromAddress("me"),
+		SessionID:  "unknown",
+		Invoice:    mockInvoice,
+	})
+	assert.Nil(t, storer.UpdateCalled)
+
+	storage.consumeSessionSpendingEvent(event.AppEventInvoicePaid{
+		ConsumerID: identity.FromAddress("me"),
+		SessionID:  mockSessionID,
+		Invoice:    mockInvoice,
+	})
+	assert.Equal(
+		t,
+		&History{
+			SessionID:         node_session.ID("sessionID"),
+			ConsumerID:        identity.FromAddress("consumerID"),
+			AccountantAddress: "0x00000000000000000000000000000000000000AC",
+			ProviderID:        identity.FromAddress("providerID"),
+			ServiceType:       "serviceType",
+			ProviderCountry:   "MU",
+			Started:           time.Date(2020, 4, 1, 10, 11, 12, 0, time.UTC),
+			Status:            "New",
+			Updated:           time.Date(2020, 4, 1, 12, 0, 0, 0, time.UTC),
+			DataStats:         connection.Statistics{},
+			Invoice:           mockInvoice,
+		},
+		storer.UpdateCalled,
+	)
 }
 
 // StubSessionStorer allows us to get all sessions, save and update them
 type StubSessionStorer struct {
 	SaveError    error
-	SaveCalled   bool
+	SaveCalled   interface{}
 	UpdateError  error
-	UpdateCalled bool
+	UpdateCalled interface{}
```

Comment:
`UpdateCalled` was good for bool. If you storing objects there it's better to use the correct name like `UpdatedObject`.